### PR TITLE
don't blow up check_perms if lsattr file doesn't exist

### LINKED ISSUE #49204: file.directory state fails if directory contains an invalid symlink
### Description of Issue/Question
If a file.directory state is run on a directory which contains a broken symlink, the state crashes.

### Setup
test state:
```
/tmp/dirtest:
  file.directory:
    - user: root
    - group: root
    - mode: 755
    - recurse:
      - user
      - group
      - mode
```

### Steps to Reproduce Issue
Success
```
# salt-call state.sls dir
local:
----------
          ID: /tmp/dirtest
    Function: file.directory
      Result: True
     Comment: The directory /tmp/dirtest is in the correct state
     Started: 10:34:25.797989
    Duration: 11.287 ms
     Changes:

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  11.287 ms
```
Create bad symlink
```
ln -s /tmp/dirtest/thing /tmp/dirtest/nope
```
State now crashes
```
# salt-call state.sls dir
/usr/local/lib/python2.7/dist-packages/requests/__init__.py:83: RequestsDependencyWarning: Old version of cryptography ([1, 2, 3]) may cause slowdown.
  warnings.warn(warning, RequestsDependencyWarning)
[ERROR   ] An exception occurred in this state: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/salt/state.py", line 1905, in call
    **cdata['kwargs'])
  File "/usr/local/lib/python2.7/dist-packages/salt/loader.py", line 1830, in wrapper
    return f(*args, **kwargs)
  File "/var/cache/salt/minion/extmods/states/file.py", line 3421, in directory
    full, ret, user, group, file_mode, None, follow_symlinks)
  File "/usr/local/lib/python2.7/dist-packages/salt/modules/file.py", line 4460, in check_perms
    lattrs = lsattr(name)
  File "/usr/local/lib/python2.7/dist-packages/salt/modules/file.py", line 564, in lsattr
    raise SaltInvocationError("File or directory does not exist.")
SaltInvocationError: File or directory does not exist.

local:
----------
          ID: /tmp/dirtest
    Function: file.directory
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/local/lib/python2.7/dist-packages/salt/state.py", line 1905, in call
                  **cdata['kwargs'])
                File "/usr/local/lib/python2.7/dist-packages/salt/loader.py", line 1830, in wrapper
                  return f(*args, **kwargs)
                File "/var/cache/salt/minion/extmods/states/file.py", line 3421, in directory
                  full, ret, user, group, file_mode, None, follow_symlinks)
                File "/usr/local/lib/python2.7/dist-packages/salt/modules/file.py", line 4460, in check_perms
                  lattrs = lsattr(name)
                File "/usr/local/lib/python2.7/dist-packages/salt/modules/file.py", line 564, in lsattr
                  raise SaltInvocationError("File or directory does not exist.")
              SaltInvocationError: File or directory does not exist.
     Started: 10:34:39.631296
    Duration: 27.711 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:  27.711 ms
```

### Versions Report
```
Salt Version:
           Salt: 2018.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.5.0
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 5.0.2
            ZMQ: 4.1.6

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-130-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

### PR DESCRIPTION
**Pull request by gtmanfred**
### What issues does this PR fix or reference?
Fixes #49204 

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by terminalmage — APPROVED** [review]

File: salt/modules/file.py [hunk 1]
```diff
@@ -4459,7 +4459,10 @@ def check_perms(name, ret, user, group, mode, attrs=None, follow_symlinks=False)
     is_dir = os.path.isdir(name)
     is_link = os.path.islink(name)
     if not salt.utils.platform.is_windows() and not is_dir and not is_link:
-        lattrs = lsattr(name)
+        try:
+            lattrs = lsattr(name)
+        except SaltInvocationError:
+            lsattrs = None
```
Comment (dgmorrisjr) [inline salt/modules/file.py]: lattrs is mis-spelled in this line.   if the except condition fires, then following `if lattrs is not None:` line will throw an `local variable 'lattrs' referenced before assignment` error.

Comment (gtmanfred) [reply salt/modules/file.py]: can you open a pr for this?

**Comment by dgmorrisjr** [comment]
I can work on that!

**Comment by dgmorrisjr** [comment]
created pull request #49827

**Pull request merged**